src/lib/containers/LoadingScreen.tsx

```diff
@@ -33,17 +35,41 @@ export const SynapseSpinner: React.FC<SynapseSpinnerProps> = ({
   )
 }
 
-type BlockingLoaderProps = {
+export type BlockingLoaderProps = {
   show: boolean
+  currentEntities?: number
+  totalEntities?: number
+  onCancel?: () => void
 }
-export const BlockingLoader: React.FC<BlockingLoaderProps> = ({ show }) => {
+export const BlockingLoader: React.FC<BlockingLoaderProps> = ({ show, currentEntities, onCancel, totalEntities }) => {
   useEffect(() => {
     document.body.style.cursor = show ? 'wait' : 'default'
     return () => {
       document.body.style.cursor = 'default'
     }
   }, [show])
+  const barLoader = (
+    <>
+      <div className="bar-loader">
+        <LinearProgress
+          variant='determinate'
+          classes={{
+            colorPrimary: 'bar-background-color',
+            barColorPrimary: 'bar-color',
+          }}
+          value={(currentEntities! / totalEntities!)*100}
 
+        />
+      </div>
+      {
+        currentEntities && 
+        <>
+          <Typography variant="headline3">Fetching selected items</Typography>
+          <Typography variant='hintText'>{currentEntities} of {totalEntities}</Typography>
```

Comment:
I would add two props `headlineText` and `hintText` and let the caller specify these.

```suggestion
          <Typography variant="headline3">{headlineText}</Typography>
          <Typography variant='hintText'>{hintText}</Typography>
```

In DetailsView I would also suggest using `currentEntities.toLocaleString()` and `totalEntities.toLocaleString()` since these are likely to be large numbers.